subsystem/kubeapi_test.go

```diff
@@ -1277,4 +1304,62 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}
 		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterReadyForInstallationCondition, controllers.ClusterNotReadyReason)
 	})
+
+	It("deploy clusterDeployment with manifest reference - bad manifest", func() {
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		ref := &corev1.LocalObjectReference{Name: "cluster-install-config"}
+		data := map[string]string{"test.yaml": "test"}
+		deployOrUpdateConfigMap(ctx, kubeClient, ref.Name, data)
+		clusterDeploymentSpec := getDefaultClusterDeploymentSNOSpec(secretRef)
+		clusterDeploymentSpec.Provisioning.ManifestsConfigMapRef = ref
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterReadyForInstallationCondition, controllers.ClusterNotReadyReason)
+
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForReconcileTimeout)
+		configureLocalAgentClient(cluster.ID.String())
+		manifests, err := agentBMClient.Manifests.ListClusterManifests(ctx, &manifests.ListClusterManifestsParams{
+			ClusterID: *cluster.ID,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(manifests.Payload)).Should(Equal(0))
+	})
+
+	It("deploy clusterDeployment with manifest reference", func() {
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		ref := &corev1.LocalObjectReference{Name: "cluster-install-config"}
+		content := `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 99-openshift-machineconfig-master-kargs
+spec:
+  kernelArguments:
+    - 'loglevel=7'`
+
+		data := map[string]string{"test.yaml": content}
+		deployOrUpdateConfigMap(ctx, kubeClient, ref.Name, data)
+		clusterDeploymentSpec := getDefaultClusterDeploymentSNOSpec(secretRef)
+		clusterDeploymentSpec.Provisioning.ManifestsConfigMapRef = ref
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterReadyForInstallationCondition, controllers.ClusterNotReadyReason)
+
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKubeName, waitForReconcileTimeout)
+		configureLocalAgentClient(cluster.ID.String())
+		Eventually(func() int {
+			manifests, err := agentBMClient.Manifests.ListClusterManifests(ctx, &manifests.ListClusterManifestsParams{
+				ClusterID: *cluster.ID,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			return len(manifests.Payload)
+		}, "1m", "2s").Should(Equal(1))
+	})
```

Comment:
What will happen if the config map is updated after the CD is created?

Reply:
it will add manifests on next reconcile.

Reply:
Who says there will be another reconcile?

Reply:
then user will miss new values from configmap. Not sure we should add watch on configmaps but we can. I will look if there is something like this in hive

Reply:
we can open ticket for it and fix it another time. Not sure when this case will be relevant to say the truth and it is relevant for all other configs we are adding